src/server/routes/continue.route.js

```diff
@@ -7,50 +7,76 @@
 const { Router } = require("express");
 const { logEmitter } = require("../services/logging.service");
 const continueController = require("../controllers/continue.controller");
+const {
+  getCouncilDataByID
+} = require("../connectors/config-db/config-db.connector");
 
 const continueRouter = () => {
   const router = Router();
 
-  router.post("/:originator", (req, res) => {
+  router.post("/:originator", async (req, res, next) => {
     logEmitter.emit(
       "functionCallWith",
       "Routes",
       "/continue route",
-      `Originator: ${req.session.council}/${req.params.originator}`
+      `Originator: ${req.params.originator}`
     );
+    try {
+      const response = continueController(
+        `/${req.params.originator}`,
+        req.session.cumulativeFullAnswers,
+        req.body,
+        req.session.switches,
+        req.session.pathConfig.path
+      );
 
-    const response = continueController(
-      `/${req.params.originator}`,
-      req.session.cumulativeFullAnswers,
-      req.body,
-      req.session.switches,
-      req.session.pathConfig.path
-    );
-
-    req.session.cumulativeFullAnswers = response.cumulativeFullAnswers;
-    req.session.validatorErrors = response.validatorErrors;
-    req.session.allValidationErrors = response.allValidationErrors;
-    req.session.switches = response.switches;
-    req.session.submissionError = [];
-    req.session.language = req.body.language;
+      req.session.cumulativeFullAnswers = response.cumulativeFullAnswers;
+      req.session.validatorErrors = response.validatorErrors;
+      req.session.allValidationErrors = response.allValidationErrors;
+      req.session.switches = response.switches;
+      req.session.submissionError = [];
+      req.session.language = req.body.language;
 
-    logEmitter.emit(
-      "functionSuccessWith",
-      "Routes",
-      "/continue route",
-      response.redirectRoute
-    );
-    req.session.save((err) => {
-      if (err) {
-        logEmitter.emit("functionFail", "Routes", "/continue route", err);
-        throw err;
+      if (req.params.originator === "la-selector") {
+        if (
+          req.body.local_authority_not_found === "yes" ||
+          req.body.local_authority === ""
+        ) {
+          response.redirectRoute = "/la-not-onboarded";
+        } else {
+          req.session.localAuthority = await getCouncilDataByID(
+            +req.body.local_authority
```

Comment:
is converting to a number enough? If it's not a number this might lead to something weird - could check in `if` above or pass straight in